### PR TITLE
docs(flutterfire_ui): Update documentation to use `currentUser` as `initialData` property when using `StreamBuilder`.

### PR DESCRIPTION
**Pull request by rubensdemelo**
add `initialData: FirebaseAuth.instance.currentUser` in  SignInScreen to avoid get `null`  in authStateChanges() even with user is signed in.

## Description

*Replace this paragraph with a description of what this PR is doing. If you're modifying existing behavior, describe the existing behavior, how this PR is changing it, and what motivated the change.*

## Related Issues

*Replace this paragraph with a list of issues related to this PR from the [issue database](https://github.com/flutter/flutter/issues). Indicate, which of these issues are resolved or fixed by this PR. Note that you'll have to prefix the issue numbers with flutter/flutter#.*

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`).
This will ensure a smooth and quick review process. Updating the `pubspec.yaml` and changelogs is not required.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [x] All existing and new tests are passing.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] The analyzer (`melos run analyze`) does not report any problems on my PR.
- [x] I read and followed the [Flutter Style Guide].
- [x] I signed the [CLA].
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change.
- [x] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/FirebaseExtended/flutterfire/blob/master/CONTRIBUTING.md
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[CLA]: https://cla.developers.google.com/

### CODE REVIEW

File: docs/ui/auth/integrating-your-first-screen.mdx [hunk 1]
```diff
@@ -118,6 +118,8 @@ Next, go ahead and return a new `SignInScreen` from the builder if the user is n
 ```dart
 return StreamBuilder<User?>(
   stream: FirebaseAuth.instance.authStateChanges(),
+  //optional
```
Comment (russellwheatley) [inline docs/ui/auth/integrating-your-first-screen.mdx:121]:
```suggestion
  // If the user is already signed-in, use it as initial data
```

File: docs/ui/auth/integrating-your-first-screen.mdx [hunk 2]
```diff
@@ -132,6 +134,8 @@ return StreamBuilder<User?>(
 );
 ```
 
+If the user is signed in, you can retrieve that data with `FirebaseAuth.instance.currentUser`. 
+
```
Comment (russellwheatley) [inline docs/ui/auth/integrating-your-first-screen.mdx:138]:
```suggestion
```

**Review by russellwheatley — CHANGES_REQUESTED** [review]
Thanks @rubensdemelo for the PR, I've tweaked a little bit.

**Comment by rubensdemelo** [comment]
> Thanks @rubensdemelo for the PR, I've tweaked a little bit.

Should I make any change ?

**Comment by rubensdemelo** [comment]
Not sure about that statement, because even user isnt signin, `initialData: FirebaseAuth.instance.currentUser` will return `false`. 

And on first build, authStateChanges returns `null`.

**Comment by rubensdemelo** [comment]
> Hey @rubensdemelo, not sure what you mean? [See here](https://github.com/FirebaseExtended/flutterfire/blob/aff72a20ce01ccb855442373b9f005dacb6f8a45/packages/firebase_auth/firebase_auth/lib/src/firebase_auth.dart#L61).
> 
> Also - the example uses a condition where it won't matter if it is `null` or `false`:
> 
> ```dart
>   if (!snapshot.hasData) {
>       return SignInScreen(
>         providerConfigs: []
>       );
>     }
> ```

Thanks

**Pull request merged**